catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/helper/RegistryPackageTypeHelper.java

```diff
@@ -57,241 +68,317 @@ public void setRegistryPackage(RegistryPackageType registryPackage) {
+     *
+     * @param registryPackage the RegistryPackageType that will be crawled, null returns an empty list
+     * @param id              to be used as the sourceObjectId of the Association
+     * @param type            Type of the object to find associations for
+     * @return a List containing the objects associated to the provided id
+     */
+    public <T extends RegistryObjectType> List<T> getAssociatedObjects(
+            RegistryPackageType registryPackage, String id, Class<T> type) {
+        if (StringUtils.isEmpty(id)) {
+            return Collections.emptyList();
+        }
+        Set<String> targetObjectIds = getAssociationTargetObjectIds(registryPackage, id);
+        if (CollectionUtils.isEmpty(targetObjectIds)) {
+            return Collections.emptyList();
+        }
+
+        return getObjectsFromRegistryObjectList(registryPackage, type).stream()
+                .filter(isAssociatedObject(targetObjectIds))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * This is a convenience method that returns all Objects associated with the provided id extracted from the provided RegistryObjectList
+     *
+     * @param registryObjectList the RegistryObjectList that will be crawled, null returns an empty list
+     * @param id                 to be used as the sourceObjectId of the Association
+     * @param type               Type of the object to find associations for
+     * @return a List containing the objects associated to the provided id
+     */
+    public <T extends RegistryObjectType> List<T> getAssociatedObjects(
+            RegistryObjectListType registryObjectList, String id, Class<T> type) {
+        if (StringUtils.isEmpty(id)) {
+            return Collections.emptyList();
+        }
+
+        Set<String> targetObjectIds = getAssociationTargetObjectIds(registryObjectList, id);
+        if (CollectionUtils.isEmpty(targetObjectIds)) {
+            return Collections.emptyList();
+        }
+
+        return getObjectsFromRegistryObjectList(registryObjectList, type).stream()
+                .filter(isAssociatedObject(targetObjectIds))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * This is a convenience method that returns all Objects associated with the provided id extracted from this class's RegistryPackageType
+     *
+     * @param id   to be used as the sourceObjectId of the Association
+     * @param type Type of the object to find associations for
+     * @return a List containing the objects associated to the provided id
+     */
+    public <T extends RegistryObjectType> List<T> getAssociatedObjects(String id, Class<T> type) {
+        if (StringUtils.isEmpty(id)) {
+            return Collections.emptyList();
+        }
+        Set<String> targetObjectIds = getAssociationTargetObjectIds(id);
+        if (CollectionUtils.isEmpty(targetObjectIds)) {
+            return Collections.emptyList();
+        }
+
```

Comment:
Yeah, I tried to pull out the common code but then ended up having different common code verify the result of the call to the commonCodePullledOut method